src/service/folders.rs

```diff
@@ -155,12 +154,25 @@ pub async fn delete_folder(org_id: &str, folder_id: &str) -> Result<HttpResponse
         )));
     }
 
-    if db::folders::get(org_id, folder_id).await.is_err() {
-        return Ok(HttpResponse::NotFound().json(MetaHttpResponse::error(
-            http::StatusCode::NOT_FOUND.into(),
-            "Dashboard folder not found".to_string(),
-        )));
+    match db::folders::exists(org_id, folder_id).await {
```

Comment:
Same comment here as in a previous comment:

> I think the previous logic may have had a subtle issue in that is assumed an `Err` response from `db::folders::get` indicates that the folder does not exist. This isn't strictly true since that function can return `Err` for a variety of reasons (eg: losing connection to the DB for some reason). Therefore I think that using `?` to fast-return an error from the call to the db function  should make this logic behave more correctly when the call to the db function happens to fail.

> I also introduced the `exists` function to make it more clear that the Boolean return value, rather then the `Ok`/`Err` status of the response itself is what communicates the existence of the folder.